### PR TITLE
Randomly pick the first URI for RoundRobinURIProvider

### PR DESCRIPTION
**Pull request by jackjlli**
Currently when the `RoundRobinURIProvider` is instantiated, the order of the list of URI doesn't change; i.e. the first URI may always be chosen for uploading/downloading.

This PR randomly picks the first URI for RoundRobinURIProvider.

### CODE REVIEW

**Comment by kishoreg** [comment]
I understand what the PR is trying to accomplish. Can you share some insight on why triggered this change?

**Comment by jackjlli** [comment]
> I understand what the PR is trying to accomplish. Can you share some insight on why triggered this change?

In some scenarios like table rebalance, there will be a burst of segment download requests that hit pinot controllers. The purpose of this change is to help distribute the download requests to all the available controllers instead of just sending requests to only 1 controller.

File: pinot-common/src/test/java/org/apache/pinot/common/utils/RoundRobinURIProviderTest.java [hunk 1]
```diff
@@ -50,128 +52,135 @@ public void testHostAddressRoundRobin()
...
-        new TestCase("http://127.0.0.1/", new String[]{"http://127.0.0.1/"}),
-        new TestCase("http://127.0.0.1/?", new String[]{"http://127.0.0.1/?"}),
-        new TestCase("http://127.0.0.1/?it=5", new String[]{"http://127.0.0.1/?it=5"}),
-        new TestCase("http://127.0.0.1/me/out?it=5", new String[]{"http://127.0.0.1/me/out?it=5"}),
-        new TestCase("http://127.0.0.1:20000", new String[]{"http://127.0.0.1:20000"}),
-        new TestCase("http://127.0.0.1:20000/", new String[]{"http://127.0.0.1:20000/"}),
-        new TestCase("http://127.0.0.1:20000/?", new String[]{"http://127.0.0.1:20000/?"}),
-        new TestCase("http://127.0.0.1:20000/?it=5", new String[]{"http://127.0.0.1:20000/?it=5"}),
-        new TestCase("http://127.0.0.1:20000/me/out?it=5", new String[]{"http://127.0.0.1:20000/me/out?it=5"}),
+        new TestCase("http://127.0.0.1", Arrays.asList("http://127.0.0.1")),
```
Comment (Jackie-Jiang) [inline pinot-common/src/test/java/org/apache/pinot/common/utils/RoundRobinURIProviderTest.java:55]: Why changing this? I do get an IDE warning when using `Arrays.asList` on single element

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-common/src/test/java/org/apache/pinot/common/utils/RoundRobinURIProviderTest.java (hunk 1 above)
Comment (jackjlli) [reply pinot-common/src/test/java/org/apache/pinot/common/utils/RoundRobinURIProviderTest.java:55]: The purpose is to leverage some APIs from `List` interface like indexOf(), which doesn't exist in array. I can update those which have IDE warning to `Collections.singletonList()` in the next push.

**Pull request merged**